src/Patterns/Parser.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace AstraticBlocks\Patterns;
+
+class Parser
+{
+  public function parsePatternData(string $content, string $metaType = '', bool $withTags = false): string
+  {
+    switch ($metaType) {
+      case 'title':
+        $openTag = "<title>";
+        $closeTag = "</title>";
+          break;
+      case 'description':
+        $openTag = "<desc>";
+        $closeTag = "</desc>";
+          break;
+      case 'categories':
+        $openTag = "<categories (.+?)>";
+        $closeTag = "</categories>";
+          break;
+      default:
+        $openTag = "<astratic-pattern>";
+        $closeTag = "</astratic-pattern>";
+          break;
+    };
+
+    preg_match("~$openTag(.+)$closeTag~misU", $content, $matches);
```

Comment:
While it works the tilde `~` operator is not that common in regex imho so I'd change it to the `/`.
Also, the open and close tags are basically the same so they could be combined into the pattern right away.

What would you say about sth like this:
```
switch ($metaType) {
      case 'title':
            $pattern = "\<title\>(.+)\<\/title\>";
          break;
      // (...)
};

preg_match("/{$pattern}/misU", $content, $matches);
// ...
```